### PR TITLE
Rendering live for draft mode along with static mode for master #328

### PR DESCRIPTION
**Pull request by pmi**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,10 +1,14 @@
 # DON'T add secrets etc here: these values will be COMMITTED to git and AVAILABLE IN THE BROWSER at runtime!
 # For non-publishable values, add and use a local file .env.local instead!
 
+API_TOKEN=%3cr3t4piK3y
```
Comment (poi33) [inline .env:4]: ? Do we want a static token in our demo? 
Should this not be generated?

File: src/components/views/Header.tsx [hunk 2]
```diff
@@ -5,50 +5,75 @@ import {getUrl} from '../../_enonicAdapter/utils';
...
-                     width={33}
-                     height={40}
-                     alt={"Enonic XP logo"}
-                />
-            )}
-        </div>
-    </header>
-)};
+            <div
+                style={{
```
Comment (poi33) [inline src/components/views/Header.tsx:21]: Put this stuff in css

File: src/components/views/Header.tsx [hunk 3]
```diff
@@ -5,50 +5,75 @@ import {getUrl} from '../../_enonicAdapter/utils';
...
+    const encPath = encodeURIComponent(getUrl(path));
+    const encToken = encodeURIComponent(process.env.NEXT_PUBLIC_API_TOKEN as string);
+    const url = `/api/revalidate?path=${encPath}&token=${encToken}`;
+    return function () {
+        fetch(url, {
+            credentials: "include"
+        })
+            .then((result) => result.json())
+            .then(json => {
+                console.info(json);
```
Comment (poi33) [inline src/components/views/Header.tsx:67]: Debug?

File: src/pages/[[...contentPath]].tsx [hunk 4]
```diff
@@ -1,64 +1,93 @@
...
-        // HTTP 404
-        notFound: (error && error.code === '404') || context.res.statusCode === 404 || catchAllInNextProdMode || undefined,
-        props: {
-            common,
-            data,
-            meta,
-            error,
-            page,
-        }
+        notFound: (error && error.code === '404') || context.res?.statusCode === 404 || catchAllInNextProdMode || undefined,
```
Comment (poi33) [inline src/pages/[[...contentPath]].tsx:52]: Optional: Could refactor into its own variable? 
Its a bit of a long one liner

File: src/pages/api/revalidate.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+export default async function handler(req: any, res: any) {
+    // Check for secret to confirm this is a valid request
+    if (req.query.token !== process.env.API_TOKEN) {
```
Comment (poi33) [inline src/pages/api/revalidate.ts:3]: Is this really how we want to do it? 
On a live server cant you read this token? So anyone could send a request this the same token and the server would approve it?

**Review by poi33 — CHANGES_REQUESTED** [review]

**Comment by poi33** [comment]
@pmi this is dependant on the pull request #338? Eg. This cant be merged before that one is.

File: src/components/views/Header.tsx (hunk 3 above)
Comment (pmi) [reply src/components/views/Header.tsx:67]: Yup )

File: .env (hunk 1 above)
Comment (pmi) [reply .env:4]: Well, that's what they advised in [nextjs docs](https://nextjs.org/docs/basic-features/data-fetching/incremental-static-regeneration#on-demand-revalidation-beta)

But we will take another try on this when we add proper security layer

File: src/components/views/Header.tsx (hunk 2 above)
Comment (pmi) [reply src/components/views/Header.tsx]: Done !

File: src/pages/[[...contentPath]].tsx (hunk 4 above)
Comment (pmi) [reply src/pages/[[...contentPath]].tsx]: Done !

**Review by poi33 — APPROVED** [review]

**Pull request merged**